src/osd/scheduler/mClockScheduler.cc

```diff
@@ -454,17 +468,61 @@ void mClockScheduler::enqueue(OpSchedulerItem&& item)
 
 void mClockScheduler::enqueue_front(OpSchedulerItem&& item)
 {
-  immediate.push_back(std::move(item));
+  unsigned priority = item.get_priority();
+  unsigned cutoff = get_io_prio_cut(cct);
+
+  if (priority >= cutoff) {
+    immediate.push_back(std::move(item));
+  } else {
+    enqueue_strict_front(priority, std::move(item));
+  }
   // TODO: item may not be immediate, update mclock machinery to permit
   // putting the item back in the queue
 }
 
+void mClockScheduler::enqueue_strict(unsigned priority, OpSchedulerItem&& item)
```

Comment:
Since there's a single high_priority queue, we can define a single `enqueue_strict(unsigned priority, OpSchedulerItem&& item, bool front=false)` and push_back or push_front the item into the high_priority_queue based on `front` flag.

Reply:
That makes sense, I'll change it.